File: crates/sui-js/examples/external-surface.js
"use strict";

const sui = require("..");

const pixels = Buffer.from([
  255, 60, 40, 255,
  40, 120, 255, 255,
  30, 180, 110, 255,
  255, 230, 80, 255,
]);
const texture = sui.ExternalTextureDescriptor.cpuRgba8(new sui.Size(2, 2), pixels, "1");
const surface = sui.ExternalSurface(
  texture,
  new sui.Size(128, 96),
  "CPU upload preview"
);

const root = sui.Column(
  [
    surface,
    sui.Label("CPU fallback external surface"),
  ],
  8
);

const window = new sui.Window("External surface");
window.root(root);

const app = new sui.App();
app.window(window);

const snapshot = app.start().render();
console.log("commands:", snapshot.commandCount);
console.log("images:", snapshot.drawImageCount, snapshot.registeredImageCount);

File: crates/sui-python/examples/external_surface.py
import sui


pixels = bytes(
    [
        255,
        60,
        40,
        255,
        40,
        120,
        255,
        255,
        30,
        180,
        110,
        255,
        255,
        230,
        80,
        255,
    ]
)
texture = sui.ExternalTextureDescriptor.cpu_rgba8(sui.Size(2, 2), pixels, generation=1)
surface = sui.ExternalSurface(
    texture,
    desired_size=sui.Size(128, 96),
    name="CPU upload preview",
)

app = sui.App()
app.window(
    sui.Window("External surface").root(
        sui.Column(
            [
                surface,
                sui.Label("CPU fallback external surface"),
            ],
            gap=8,
        )
    )
)

snapshot = app.start().render()
print("commands:", snapshot.command_count)
print("images:", snapshot.draw_image_count, snapshot.registered_image_count)
